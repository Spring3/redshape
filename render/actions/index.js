import issueActions from './issue.actions';
import timeEntryActions from './timeEntry.actions';
import settingsActions from './settings.actions';

export default {
  issue: issueActions,
  timeEntry: timeEntryActions,
  settings: settingsActions
};
