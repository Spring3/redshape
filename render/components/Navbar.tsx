import React, { useCallback } from 'react';
import { useTheme } from 'styled-components';
import { NavLink, useMatch } from 'react-router-dom';
import { css } from '@emotion/react';
import ChevronDownIcon from 'mdi-react/ChevronDownIcon';
import ChevronUpIcon from 'mdi-react/ChevronUpIcon';

import { GhostButton } from './GhostButton';
import { useOvermindActions, useOvermindState } from '../store';
import { Flex } from './Flex';
import { Dropdown } from './Dropdown';
import { theme as Theme } from '../theme';
import { useNavbar } from '../contexts/NavbarContext';
import { BackButton } from './BackButton';

const styles = {
  nav: css`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    z-index: 2;
    height: 50px;
    background: white;
    display: flex;
    align-items: center;
    justify-content: space-between;
    box-shadow: 0px 0px 5px 0px lightgrey;
    padding: 20px 1.5rem 10px 1.5rem;
  `,
  icon: css`
    vertical-align: middle;
  `
};

const Navbar = () => {
  const navbarState = useNavbar();
  const state = useOvermindState();
  const actions = useOvermindActions();

  const userName = `${state.users.currentUser?.firstName} ${state.users.currentUser?.lastName}`;

  const theme = useTheme() as typeof Theme;

  const isIssuesPage = Boolean(useMatch('/issues/'));

  const navLinkStyles = css`
    font-size: 1rem;
    text-decoration: none;
    font-weight: semibold;
    border-bottom: 2px solid transparent;
    color: ${theme.normalText};
    transition: color ease ${theme.transitionTime};
    padding-bottom: 5px;

    &:active {
      color: ${theme.main};
    }

    &:hover {
      cursor: pointer;
      color: ${theme.main};
      border-bottom: 2px solid ${theme.main};
    }
  `;

  const getDropdownToggleElement = useCallback(({ toggle, isOpen }) => (
    <a css={navLinkStyles} onClick={toggle} href="#">
      {userName}
      {' '}
      {isOpen ? <ChevronUpIcon size={20} css={styles.icon} /> : <ChevronDownIcon size={20} css={styles.icon} />}
    </a>
  ), []);

  return (
    <nav css={styles.nav}>
      <Flex alignItems="center">
        {!isIssuesPage ? <BackButton /> : null}
        <h2>{navbarState.title}</h2>
      </Flex>
      <Flex gap="1rem">
        <NavLink
          css={[navLinkStyles, css`margin-right: 1rem`]}
          to="/issues"
          style={({ isActive }) => {
            if (isActive) {
              return ({
                color: theme.main,
                borderBottom: `2px solid ${theme.main}`
              });
            }
            return {};
          }}
        >
          Backlog

        </NavLink>
        <Dropdown getDropdownToggleElement={getDropdownToggleElement}>
          <GhostButton fullWidth id="signout" onClick={actions.users.logout}>
            Sign out
          </GhostButton>
        </Dropdown>
      </Flex>
    </nav>
  );
};

export {
  Navbar
};
