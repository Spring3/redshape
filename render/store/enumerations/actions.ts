import type { IAction, Context } from 'overmind';
import { Activity, Response } from '../../../types';

const fetchActivities: IAction<void, Promise<Response<Activity[]>>> = async ({ effects, state }: Context) => {
  const response = await effects.mainProcess.request({
    payload: {
      method: 'GET',
      route: 'enumerations/time_entry_activities.json',
    }
  });

  if (response.success) {
    state.enumerations.activities = response.data.activities as Activity[] ?? [];
    return {
      success: true,
      data: state.enumerations.activities,
    };
  }

  return {
    success: false,
    error: response.error
  };
};

export {
  fetchActivities
};
