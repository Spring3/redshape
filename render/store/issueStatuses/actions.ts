import { IAction, Context } from 'overmind';
import { IssueStatus, PaginatedActionResponse } from '../../../types';
import { indexById } from '../helpers';

const getAll: IAction<void, Promise<PaginatedActionResponse<IssueStatus>>> = async ({ effects, state }: Context) => {
  const response = await effects.mainProcess.request({
    payload: {
      method: 'GET',
      route: 'issue_statuses.json'
    }
  });

  if (response.success) {
    state.issueStatuses.byId = {
      ...state.issueStatuses,
      ...indexById(response.payload.issueStatuses)
    };

    return {
      success: true,
      data: {
        items: response.payload.issueStatuses,
        total: response.payload.total,
        limit: response.payload.limit,
        offset: response.payload.offset
      },
      hasMore: response.payload.total > (response.payload.offset + response.payload.issueStatuses.length)
    };
  }

  return {
    success: false,
    data: {
      items: [],
      total: 0,
      limit: response.payload.limit,
      offset: response.payload.offset
    },
    hasMore: false,
    error: response.error
  };
};

export {
  getAll
};
