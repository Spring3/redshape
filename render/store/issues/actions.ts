import type { Context, IAction } from 'overmind';
import { Issue, PaginatedActionResponse, Response } from '../../../types';
import { indexById } from '../helpers';

type GetManyIssuesArgs = {
  filters: {
    assigned_to_id?: number | string;
    status_id?: '*' | 'open' | 'closed';
    subject?: string;
    sort?: string;
  },
  offset?: number;
  limit?: number;
}

const getMany: IAction<GetManyIssuesArgs, Promise<PaginatedActionResponse<Issue>>> = async ({ effects, state }: Context, { filters, offset = 0, limit = 20 }) => {
  const query = {
    limit: limit ? Math.abs(limit) : undefined,
    offset: offset ? Math.abs(offset) : 0,
    include: 'attachments,relations',
    ...filters
  };

  const response = await effects.mainProcess.request({
    payload: {
      route: 'issues.json',
      method: 'GET',
      query
    }
  });

  if (response.success) {
    state.issues.byId = {
      ...state.issues.byId,
      ...indexById(response.payload.issues)
    };

    console.log('payload', response.payload);

    return {
      success: true,
      data: {
        items: response.payload.issues,
        total: response.payload.total,
        limit: response.payload.limit,
        offset: response.payload.offset
      },
      hasMore: response.payload.total > (response.payload.offset + response.payload.issues.length),
    };
  }

  return {
    success: false,
    data: {
      items: [],
      total: 0,
      limit,
      offset
    },
    hasMore: false,
    error: response.error
  };
};

type GetOneIssueArgs = {
  id: string;
}

const getOne: IAction<GetOneIssueArgs, Promise<Response<Issue>>> = async ({ effects, state }: Context, { id }) => {
  const response = await effects.mainProcess.request({
    payload: {
      route: `issues/${id}.json`,
      method: 'GET',
      query: {
        include: 'attachments,children,relations,journals',
      }
    }
  });

  if (response.success) {
    const issue = response.payload as Issue;
    state.issues.byId[id] = issue;
    return {
      success: true,
      data: issue
    };
  }

  return {
    success: false,
    error: response.error
  };
};

export {
  getOne,
  getMany
};
