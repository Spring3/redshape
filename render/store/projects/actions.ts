import type { IAction, Context } from 'overmind';
import { PaginatedActionResponse } from '../../../types';
import { indexById } from '../helpers';

type GetManyProjectsArgs = {
  offset?: number;
  limit?: number;
}

const getManyProjects: IAction<GetManyProjectsArgs, Promise<PaginatedActionResponse<any>>> = async ({ state, effects }: Context, { offset = 0, limit = 20 }) => {
  const normalizedOffset = offset ? Math.abs(offset) : 0;
  const normalizedLimit = limit ? Math.abs(limit) : undefined;

  const response = await effects.mainProcess.request({
    payload: {
      route: 'projects.json',
      method: 'GET',
      query: {
        offset: normalizedOffset,
        include: 'time_entry_activities',
        limit: normalizedLimit
      }
    }
  });

  if (response.success) {
    state.projects.byId = {
      ...state.projects.byId,
      ...indexById(response.data.projects)
    };

    console.log('payload', response.data);

    return {
      success: true,
      data: {
        items: response.data.projects,
        total: response.data.total,
        limit: response.data.limit,
        offset: response.data.offset
      },
      hasMore: response.data.total > response.data.projects.length
    };
  }

  return {
    success: false,
    data: {
      items: [],
      total: 0,
      limit: response.data.limit,
      offset: response.data.offset
    },
    hasMore: false,
    error: response.error
  };
};

export {
  getManyProjects
};
