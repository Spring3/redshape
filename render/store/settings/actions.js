const updateSettings = ({ state }) => {
};

export {
  updateSettings
};
