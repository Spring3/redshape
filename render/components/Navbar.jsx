import React, { Component } from 'react';
import styled, { css, withTheme } from 'styled-components';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';


import ArrowLeftIcon from 'mdi-react/ArrowLeftIcon';
import RefreshIcon from 'mdi-react/RefreshIcon';
import LogoutIcon from 'mdi-react/LogoutIcon';
import GearIcon from 'mdi-react/GearIcon';
import PlaylistCheckIcon from 'mdi-react/PlaylistCheckIcon';
import PlaylistPlusIcon from 'mdi-react/PlaylistPlusIcon';
import PlaylistStarIcon from 'mdi-react/PlaylistStarIcon';
import ViewHeadlineIcon from 'mdi-react/ViewHeadlineIcon';
import { animationSlideRight } from '../animations';
import { GhostButton } from './Button';
import actions from '../actions';

import Modal from './Modal';
import AboutPage from '../about/AboutPage';

const Name = styled.span`
  color: ${props => props.theme.main};
  align-self: flex-end;
`;

const Bubble = styled.div`
  .Label {
    width: 38px;
    height: 38px;
    border-radius: 50%;
    font-size: 18px;
    color: #fff;
    background-color: ${props => props.theme.main};
    text-align: center;
    line-height: 38px;
    font-weight: bold;
  }
  .Drop ul{
      padding: 0;
      list-style: none;
  }

  .Drop ul li{
      display: inline-block;
      position: relative;
      line-height: 21px;
      text-align: left;
  }

  .Drop ul.dropdown{
      background: white;
      position: absolute;
      z-index: 999;
      visibility: hidden;
      opacity: 0;
      overflow: hidden;
      height: 0;

      ${props => (props.withAvatar ? css`
      right: 35px;
      margin-top: 11px;
      ` : css`
      right: 30px;
      margin-top: 16px;
      `)}
      margin-left: -150px;
      height: 150px;
      border: 2px solid ${props => props.theme.main};

      display: flex;
      flex-direction: column;
      transition: height 0.8s ease-out, opacity 300ms;
      -webkit-transition: height 0.8s ease-out, opacity 300ms;
      border-radius: 0 0 5px 5px;
  }

  .Drop:hover ul.dropdown{
      overflow: initial;
      visibility: visible;
      opacity: 1;
      height: auto;
  }

  .Drop ul.dropdown li{
      display: block;
      margin: 0;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 40px;
      margin: 0 30px;
      &:nth-child(2) {
        height: 60px;
      }
      &.User {
        &.User-avatar {
          margin-top: 35px;
          margin-bottom: 20px;
        }
      }
  }
  .Drop ul.dropdown li.name {
    align-self: flex-end;
  }
  .Drop ul.dropdown li.dropdown-ghost{
    height: 40px;
    position: absolute;
    width: 100%;
    top: -40px;
    margin: 0;

    &::after {
      content: ' ';
      position: absolute;
      // width: auto;
      border: 2px solid ${props => props.theme.main};
      border-width: 12px;
      border-color: ${props => props.theme.main} transparent transparent transparent;
      transform: rotate(180deg);
      right: 14px;
      bottom: 0;
    }
  }
`;

const Navbar = styled.nav`
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  z-index: 2;
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  ${props => (props.isEnhanced ? css`
    padding: 10px 40px;
    background: white;
    border-bottom: 2px solid ${props.theme.main};
    box-shadow: 0px 2px 20px #F9F9F9;
  ` : css`
    padding: 20px 40px 0px 40px;
    background: linear-gradient(to bottom, ${props => props.theme.bg} 85%, transparent);
  `)}

  ul {
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: flex;
    align-items: center;

    li {
      display: inline;
      font-size: 15px;
      font-weight: bold;
      ${({ theme }) => css`
        color: ${theme.normalText};
        transition: color ease ${theme.transitionTime};

        a {
          text-decoration: none;
          color: ${theme.normalText};
          transition: color ease ${theme.transitionTime};
          padding-bottom: 5px;
        }

        a.active {
          color: ${theme.main};
          border-bottom: 2px solid ${theme.main};
        }

        &:hover {
          a {
            cursor: pointer;
            color: ${theme.main};
          }
        }
      `}
    }
  }

  ul:first-child {
    li {
      margin: 0px 20px;
    }

    li:first-child {
      margin-left: 0;
    }
  }

  ${props => (!props.isEnhanced) && css`
  ul:last-child {
    li {
      margin: 0px 20px;
    }

    li:last-child {
      margin-right: 0px;
    }
  }
  `}
`;

const IconButton = styled(GhostButton)`
  svg {
    border-radius: 3px;
    ${({ theme }) => css`
      color: ${theme.main};
      border: 2px solid transparent;
      transition: all ease ${theme.transitionTime};

      &:hover {
        color: ${theme.main};
        border: 2px solid ${theme.main};
      }
    `}
  }
`;

const IconMovingButton = styled(IconButton)`
  svg {
    animation: ${animationSlideRight} 2s ease-in infinite;
    &:hover {
      animation-play-state: paused;
    }
  }
`;

const User = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  span:nth-child(2) {
    margin-top: 5px;
  }
`;
const Avatar = styled.img`
  display: block;
  border-radius: 5px;
`;

class NavigationBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpenAbout: false,
      allowAvatar: true
    };
  }

  signout = () => {
    const { logout } = this.props;
    logout();
  }

  checkAvatar = ({ target: img }) => {
    const fallback = !(img && img.complete && img.naturalWidth > 0);
    if (fallback) {
      this.setState({ allowAvatar: false });
      const detail = img && {
        src: `${(img.src).slice(0, 50)}...`, srcLength: img.src.length, width: img.width, height: img.height, naturalWidth: img.naturalWidth, complete: img.complete
      };
      this.props.addLog({ message: 'Avatar cannot be shown', detail });
    }
  }

  render() {
    const {
      user = {}, uiStyle, history, onRefresh
    } = this.props;
    const { name } = user;
    const { allowAvatar } = this.state;
    const isEnhanced = uiStyle === 'enhanced';
    let avatar;
    if (isEnhanced && allowAvatar) {
      avatar = user.avatar;
    }
    return (
      <Navbar isEnhanced={isEnhanced}>
        <ul>
          <li>
            <div>
              { isEnhanced ? (
                <IconButton onClick={() => history.goBack()}>
                  <ArrowLeftIcon size={30} />
                </IconButton>
              ) : (
                <IconMovingButton onClick={() => history.goBack()}>
                  <ArrowLeftIcon size={30} />
                </IconMovingButton>
              )}
              {
                isEnhanced && (
                  <IconButton onClick={onRefresh}>
                    <RefreshIcon size={30} />
                  </IconButton>
                )
              }
            </div>
          </li>
          <li>
            <NavLink to="/app/summary/assigned">
              { isEnhanced && (<PlaylistCheckIcon size={22} style={{ marginTop: -2, marginRight: 4 }} />)}
              Assigned
            </NavLink>
          </li>
          <li>
            <NavLink to="/app/summary/author">
              { isEnhanced && (<PlaylistPlusIcon size={22} style={{ marginTop: -2, marginRight: 4 }} />)}
              Author
            </NavLink>
          </li>
          <li>
            <NavLink to="/app/summary/watching">
              { isEnhanced && (<PlaylistStarIcon size={22} style={{ marginTop: -2, marginRight: 4 }} />)}
              Watching
            </NavLink>
          </li>
          <li>
            <NavLink to="/app/summary/all">
              { isEnhanced && (<ViewHeadlineIcon size={18} style={{ height: 22, marginTop: -2, marginRight: 4 }} />)}
              All
            </NavLink>
          </li>
        </ul>
        { isEnhanced ? (
          <Bubble withAvatar={!!avatar}>
            <div className="Drop">
              {
                avatar != null ? (
                  <Avatar
                    onError={this.checkAvatar}
                    onLoad={this.checkAvatar}
                    width={avatar.size}
                    height={avatar.size}
                    src={avatar.img}
                  />
                ) : (
                  <div className="Label">{name && name.charAt(0) || 'Me'}</div>
                )
              }
              <ul className="dropdown">
                <li className="dropdown-ghost" />
                <li>
                  <NavLink to="/app/settings">
                    { isEnhanced && (<GearIcon size={18} style={{ marginTop: -2, marginRight: 2 }} />)}
                    Settings
                  </NavLink>
                </li>
                <li>
                  <GhostButton onClick={() => this.setState({ isOpenAbout: true })}>About Redshape</GhostButton>
                </li>
                <li className={avatar ? 'User User-avatar' : 'User'}>
                  <User>
                    {
                      avatar != null && (<div className="Label">{name.charAt(0)}</div>)
                    }
                    <Name>{name}</Name>
                  </User>
                </li>
                <li>
                  <GhostButton
                    id="signout"
                    onClick={this.signout}
                  >
                    {isEnhanced && (<LogoutIcon size={18} style={{ marginTop: -2, marginRight: 2 }} />)}
                    Sign out
                  </GhostButton>
                </li>
              </ul>
            </div>
          </Bubble>
        ) : (
          <ul>
            <li>
              <NavLink to="/app/settings">
                { isEnhanced && (<GearIcon size={18} style={{ marginTop: -2, marginRight: 2 }} />)}
                Settings
              </NavLink>
            </li>
            <li>
              <Name>{name}</Name>
            </li>
            <li>
              <GhostButton
                id="signout"
                onClick={this.signout}
              >
                {isEnhanced && (<LogoutIcon size={18} style={{ marginTop: -2, marginRight: 2 }} />)}
                Sign out
              </GhostButton>
            </li>
          </ul>
        )}
        <Modal
          open={!!this.state.isOpenAbout}
          onClose={() => this.setState({ isOpenAbout: false })}
          center={true}
        >
          <AboutPage modal={true} />
        </Modal>
      </Navbar>
    );
  }
}

NavigationBar.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
    name: PropTypes.string.isRequired,
    api_key: PropTypes.string.isRequired,
    redmineEndpoint: PropTypes.string.isRequired
  }).isRequired,
  logout: PropTypes.func.isRequired,
  uiStyle: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  user: state.user,
  uiStyle: state.settings.uiStyle,
});

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(actions.user.logout()),
  fetchAvatar: id => dispatch(actions.user.fetchAvatar(id)),
  addLog: reg => dispatch(actions.session.addLog(reg)),
});

export default withRouter(withTheme(connect(mapStateToProps, mapDispatchToProps)(NavigationBar)));
