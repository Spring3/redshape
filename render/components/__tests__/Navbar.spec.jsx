import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { HashRouter, Route } from 'react-router-dom';
import thunk from 'redux-thunk';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { USER_LOGOUT } from '../../actions/user.actions';
import Navbar from '../Navbar';

const mockStore = configureStore([thunk]);

describe('Navbar component', () => {
  afterEach(cleanup);
  it('should match the snapshot', () => {
    const state = {
      user: {
        id: 1,
        name: 'Anonymous',
        api_key: 1
      }
    };
    const store = mockStore(state);
    const { getByText } = render(
      <Provider store={store}>
        <HashRouter>
          <Route path="/" component={(props) => <Navbar {...props} />} />
        </HashRouter>
      </Provider>
    );
    const summary = getByText('Summary');
    expect(summary).toBeDefined();
    expect(summary).toHaveAttribute('href', '#/app/summary');
    const user = getByText(state.user.name);
    expect(user).toBeDefined();
    expect(user).toHaveAttribute('href', '#/app/summary');
    expect(getByText('Sign out')).toBeDefined();
  });

  it('should fire a logout action when the signout button is clicked', () => {
    const store = mockStore({
      user: {
        id: 1,
        name: 'Anonymous',
        api_key: 1
      }
    });
    render(
      <Provider store={store}>
        <HashRouter>
          <Route path="/" component={(props) => <Navbar {...props} />} />
        </HashRouter>
      </Provider>
    );


    const signoutBtn = document.querySelector('#signout');
    expect(signoutBtn).toBeDefined();
    fireEvent.click(signoutBtn);
    expect(store.getActions().length).toBe(2);
    expect(store.getActions().pop()).toEqual({
      type: USER_LOGOUT
    });
  });
});
