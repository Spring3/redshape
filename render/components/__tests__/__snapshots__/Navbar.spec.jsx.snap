// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match the snapshot 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color ease;
  transition: color ease;
}

.c1:active,
.c1:focus,
.c1:visited {
  background: transparent;
}

.c0 {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  z-index: 2;
  height: 50px;
  background: linear-gradient(to bottom,85%,transparent);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px 40px 0px 40px;
}

.c0 ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 ul li {
  display: inline;
  font-size: 15px;
  font-weight: bold;
  -webkit-transition: color ease;
  transition: color ease;
}

.c0 ul li a {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color ease;
  transition: color ease;
  padding-bottom: 5px;
}

.c0 ul li a.active {
  border-bottom: 2px solid;
}

.c0 ul li:hover {
  cursor: pointer;
}

.c0 ul:first-child li {
  margin: 0px 20px;
}

.c0 ul:first-child li:first-child {
  margin-left: 0;
}

.c0 ul:last-child li {
  margin: 0px 20px;
}

.c0 ul:last-child li:last-child {
  margin-right: 0px;
}

<nav
  className="c0"
>
  <ul>
    <li>
      <a
        aria-current={null}
        href="#/app/summary"
        onClick={[Function]}
      >
        Summary
      </a>
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="#/app/summary"
        onClick={[Function]}
      >
        Anonymous
      </a>
    </li>
    <li>
      <a
        className="c1"
        disabled={false}
        href="#"
        id="signout"
        onClick={[Function]}
      >
        Sign out
      </a>
    </li>
  </ul>
</nav>
`;
